### PR TITLE
Add parameter to allow or not a generator to scale when its initial targetP is outside minP - maxP

### PR DESCRIPTION
**Pull request by FranckLecuyer**
**Please check if the PR fulfills these requirements**

- [ ] The commit message follows our guidelines
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**Does this PR already have an issue describing the problem?**

**What kind of change does this PR introduce?**

Feature

**What is the current behavior?**

A generator cannot scale when it's intial targetP value is less than the minimum value or is greater than the maximum value.

**What is the new behavior (if this is a feature change)?**
A boolean parameter 'allowOutOfBoundsTargetP' allow the generator to scale when it's intial targetP value is less than the minimum value or is greater than the maximum value.

**Does this PR introduce a breaking change or deprecate an API?**

- [ ] The *Breaking Change* or *Deprecated* label has been added
- [ ] The migration guide has been updated in the github wiki *(What changes might users need to make in their application due to this PR?)*

**Other information**:

### CODE REVIEW

File: iidm/iidm-modification/src/main/java/com/powsybl/iidm/modification/scalable/GeneratorScalable.java [hunk 1]
```diff
@@ -24,6 +24,8 @@ class GeneratorScalable extends AbstractInjectionScalable {
 
     private static final Logger LOGGER = LoggerFactory.getLogger(GeneratorScalable.class);
 
+    private boolean allowOutOfBoundsTargetP = false;
```
Comment (annetill) [inline iidm/iidm-modification/src/main/java/com/powsybl/iidm/modification/scalable/GeneratorScalable.java:27]: Can you use the `ScalingParameters` class to define the parameter?

**Review by annetill — CHANGES_REQUESTED** [review]

File: iidm/iidm-modification/src/main/java/com/powsybl/iidm/modification/scalable/GeneratorScalable.java (hunk 1 above)
Comment (FranckLecuyer) [reply iidm/iidm-modification/src/main/java/com/powsybl/iidm/modification/scalable/GeneratorScalable.java]: Yes. I have change this

File: iidm/iidm-modification/src/main/java/com/powsybl/iidm/modification/scalable/GeneratorScalable.java [hunk 2]
```diff
@@ -126,7 +126,7 @@ public double scale(Network n, double asked, ScalingParameters parameters) {
         double oldTargetP = g.getTargetP();
         double minimumTargetP = minimumTargetP(g);
         double maximumTargetP = maximumTargetP(g);
-        if (oldTargetP < minimumTargetP || oldTargetP > maximumTargetP) {
+        if (!parameters.isAllowOutOfBoundsGeneratorTargetP() && (oldTargetP < minimumTargetP || oldTargetP > maximumTargetP)) {
```
Comment (annetill) [inline iidm/iidm-modification/src/main/java/com/powsybl/iidm/modification/scalable/GeneratorScalable.java:129]: After a talk with @geofjamg we prefer `parameters.allowsGeneratorOutOfActivePowerLimits()`.

**Review by annetill — APPROVED** [review]

**Review by geofjamg — APPROVED** [review]

File: iidm/iidm-modification/src/main/java/com/powsybl/iidm/modification/scalable/GeneratorScalable.java (hunk 2 above)
Comment (FranckLecuyer) [reply iidm/iidm-modification/src/main/java/com/powsybl/iidm/modification/scalable/GeneratorScalable.java]: Renamed

**Pull request merged**